src/Microsoft.Health.CosmosDb/Exceptions/CosmosDbException.cs

```diff
@@ -0,0 +1,14 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.CosmosDb.Exceptions
+{
+    public class CosmosDbException : Exception
+    {
+        public string CustomExceptionMessage { get; protected set; }
```

Comment:
We should reuse the existing Message property instead of introduce a new one (refer MethodNotAllowedException for implementation).